### DISCUSSION TITLE
Feature Request:  job.<job id>.on

### DISCUSSION THREAD

**Discussion by AlanKuurstra** [community/community#25571] [Actions]
<p>I would like to have a workflow with 2 jobs.  The first job should run on every branch push.  The second job depends on the first job and should only be run on a master branch push.  This would require the <a href="https://help.github.com/en/actions/automating-your-workflow-with-github-actions/workflow-syntax-for-github-actions#on" rel="nofollow noopener">on (https://help.github.com/en/actions/automating-your-workflow-with-github-actions/workflow-syntax-for-github-actions#on)</a> functionality to work at the job level.  </p>
<p>This could also be solved if I could make a workflow depend on other workflows.  That is, if the <a href="https://help.github.com/en/actions/automating-your-workflow-with-github-actions/workflow-syntax-for-github-actions#jobsjob_idneeds" rel="nofollow noopener">need (https://help.github.com/en/actions/automating-your-workflow-with-github-actions/workflow-syntax-for-github-actions#jobsjob_idneeds)</a> functionality worked at the workflow level.</p>

**Comment by BradenM** [comment, marked as answer]
<p>Hi <a class="mention" href="/alankuurstra">@alankuurstra</a>!</p>
<p>This is actually currently possible using <a href="https://help.github.com/en/actions/automating-your-workflow-with-github-actions/workflow-syntax-for-github-actions#jobsjob_idif" rel="nofollow noopener">jobs.&lt;job id&gt;.if</a> conditional with the <a href="https://help.github.com/en/actions/automating-your-workflow-with-github-actions/contexts-and-expression-syntax-for-github-actions#github-context" rel="nofollow noopener">github.ref</a> context item.</p>
<p>Here is an example:</p>
<pre><code>jobs:
  job1:
    steps:
      - run: echo "I always execute!"
  job2:
    if: github.ref == 'refs/heads/master'
    steps:
      - run: echo "I only execute on master branch!"
job3:
if: contains(github.ref, 'release')
steps:
- run: echo "I only run if the pushed branch contains 'release' in its name!
</code></pre>

**Comment by AlanKuurstra** [comment]
<p>Thanks for the work around!</p>
